apis/core-api/src/core/services/FanClubService.ts

```diff
@@ -0,0 +1,173 @@
+import { FanClubMemberRoleEnum, FanClubMemberStatusEnum } from '@ultras/utils';
+import db from 'core/data/models';
+import { Optional } from 'sequelize';
+import { DbIdentifier } from 'types';
+import { FanClub, FanClubCreationAttributes } from 'core/data/models/FanClub';
+
+import BaseService from './BaseService';
+
+interface MembershipInterface {
+  fanClubId: DbIdentifier;
+  userId: DbIdentifier;
+}
+interface WithRoleInterface {
+  role: FanClubMemberRoleEnum;
+}
+interface WithStatusInterface {
+  status: FanClubMemberStatusEnum;
+}
+
+interface CreateMemberInterface
+  extends MembershipInterface,
+    WithRoleInterface,
+    WithStatusInterface {}
+
+class FanClubService extends BaseService {
+  static async create({
+    name,
+    description,
+    cityId,
+    countryId,
+    teamId,
+    avatar,
+    coverPhoto,
+    privacy,
+  }: FanClubCreationAttributes) {
+    const fanClub = await db.FanClub.create({
+      name,
+      description,
+      cityId,
+      countryId,
+      teamId,
+      avatar,
+      coverPhoto,
+      privacy,
+    });
+
+    return fanClub;
+  }
+
+  static async update(
+    id: DbIdentifier,
+    {
+      name,
+      description,
+      cityId,
+      countryId,
+      avatar,
+      coverPhoto,
+      privacy,
+    }: Partial<Omit<FanClubCreationAttributes, 'teamId'>>
+  ) {
+    const fanClub = await db.FanClub.findOne({
+      where: {
+        id: id,
+      },
+    });
+
+    if (typeof name != 'undefined') {
```

Comment:
Why we need hose checks ?

Reply:
It can call a partial update, so I'm just updating fields that are provided from the client-side.